components/brave_rewards/browser/database/publisher_info_database.cc

```diff
@@ -1587,6 +1587,78 @@ bool PublisherInfoDatabase::MigrateV6toV7() {
   return transaction.Commit();
 }
 
+bool PublisherInfoDatabase::MigrateV7toV8() {
+  sql::Transaction transaction(&GetDB());
+  if (!transaction.Begin()) {
+    return false;
+  }
+
+  if (!MigrateToV8ContributionInfoTable()) {
+    return false;
+  }
+
+  if (!MigrateToV8PendingContributionsTable()) {
+    return false;
+  }
+
+  return transaction.Commit();
+}
+
+bool PublisherInfoDatabase::MigrateToV8ContributionInfoTable() {
+  std::string table_name = "contribution_info";
+  std::string temp_table_name = "contribution_info_temp";
+
+  if (!RenameDBTable(table_name, temp_table_name)) {
+    return false;
+  }
+
+  if (!CreateContributionInfoTable()) {
```

Comment:
this will not work as you are creating table from the latest version. For example if you migrate from 7 to 9, you will create here table of 9 instead of version 8